siad/network.go

```diff
@@ -92,16 +151,19 @@ func (c *Core) SendBlocks(knownBlocks [32]consensus.BlockID) (blocks []consensus
 // progressing exponentially backwards to the genesis block. The receiver uses
 // these blocks to find the most recent block seen by both peers, and then
 // transmits blocks sequentially until the requester is fully synchronized.
-func (c *Core) CatchUp(peer network.Address) {
+func (d *daemon) CatchUp(peer network.Address) {
 	var newBlocks []consensus.Block
-	err := peer.RPC("SendBlocks", c.blockHistory(), &newBlocks)
+	err := peer.RPC("SendBlocks", d.blockHistory(), &newBlocks)
 	if err != nil && err.Error() != moreBlocksErr.Error() {
 		// log error
 		// TODO: try a different peer?
 		return
 	}
 	for _, block := range newBlocks {
-		c.AcceptBlock(block)
+		err = d.state.AcceptBlock(block)
+		if err != nil {
+			// TODO: something
+		}
```

Comment:
if an error is returned, it means that there's a problem with the block, and that that block will always return an error.

(actually, not strictly true, the block might be a FutureBlock, which means you need to handle it differently).

But mostly it means you should stop catching up from the person you're currently asking.